### PR TITLE
Add module benchmark

### PR DESCRIPTION
**Pull request by sangjinhan**
#130 introduced significant performance degradation: e.g., the throughput of samples/s2s.bess dropped about 35%. Before working on optimization, this PR adds benchmark tests regarding invocation of module code.

### CODE REVIEW

File: core/worker.cc [hunk 1]
```diff
@@ -20,7 +20,9 @@
 int num_workers = 0;
 std::thread worker_threads[MAX_WORKERS];
 Worker *volatile workers[MAX_WORKERS];
-thread_local Worker ctx;
+
+// See worker.h
+__thread Worker ctx;
```
Comment (melvinw) [inline core/worker.cc]: Why this change?

File: core/worker.h [hunk 2]
```diff
@@ -45,24 +45,6 @@ struct gate_task {
...
-        silent_drops_(),
-        current_tsc_(),
-        current_ns_(),
-        igate_stack_(),
-        stack_depth_(),
-        pending_gates_(),
-        splits_() {}
-
-  virtual ~Worker() {}
-
```
Comment (melvinw) [inline core/worker.h]: Ah. Nevermind

File: core/worker.cc (hunk 1 above)
Comment (sangjinhan) [inline core/worker.cc]: worker.h has some brief comments on it. One concern is __thread is a gcc extension, but seems universally supported by other compilers as well.
```
// NOTE: Do not use "thread_local" here. It requires a function call every time
// it is accessed. Use __thread instead, which incurs minimal runtime overhead.
// Worker must be "trivial" type.
```

**Comment by sangjinhan** [comment]
Build fails since it lacks a change I was about to push. Will hold for now.

File: core/worker.h [hunk 3]
```diff
@@ -167,25 +135,26 @@ class Worker {
...
 };
 
 extern int num_workers;
 extern std::thread worker_threads[MAX_WORKERS];
 extern Worker *volatile workers[MAX_WORKERS];
-extern thread_local Worker ctx;
+
+// NOTE: Do not use "thread_local" here. It requires a function call every time
+// it is accessed. Use __thread instead, which incurs minimal runtime overhead.
+// Worker must be "trivial" type.
```
Comment (melvinw) [inline core/worker.h]: might be worthwhile to static assert on std::is_trivial (http://en.cppreference.com/w/cpp/types/is_trivial) here

Comment (sangjinhan) [inline core/worker.h]: Yup, will do
